src/utils/cairoTypeSystem.ts

```diff
@@ -169,6 +169,24 @@ export class CairoFelt extends CairoType {
   }
 }
 
+export class CairoUint extends CairoType {
+  constructor(public nBits: number = 256) {
+    super();
+  }
+  get fullStringRepresentation(): string {
+    return `[u${this.nBits}]`;
+  }
+  toString(): string {
+    return `u${this.nBits}`;
+  }
+  get width(): number {
+    return 1; // not sure about this width, but for the moment consider it as a felt
```

Comment:
Where is the width used? I supposed when e.g. iterating over an array? I suppose it will always be 1 in cairo 1.0

Reply:
I suppose it is used to as a felt slot in memory and storage systems

Reply:
width represents felt size of the cairo type. For example a tuple of three felts has size 3, a tuple of three uint256 has size 6 etc. 

Reference types usually have size one representing their pointer to somewhere in memory or storage. But that can vary depending on the type and datalocation.